optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+        "default": "distilbert-base-cased",
+        "type": "text",  # feature extraction is only supported for text at the moment
+    },
+    "fill-mask": {
+        "impl": FillMaskPipeline,
+        "class": (NeuronModelForMaskedLM,),
+        "default": "bert-base-cased",
+        "type": "text",
+    },
+    "question-answering": {
+        "impl": QuestionAnsweringPipeline,
+        "class": (NeuronModelForQuestionAnswering,),
+        "default": "distilbert-base-cased-distilled-squad",
+        "type": "text",
+    },
+    "text-classification": {
+        "impl": TextClassificationPipeline,
+        "class": (NeuronModelForSequenceClassification,),
+        "default": "distilbert-base-uncased-finetuned-sst-2-english",
+        "type": "text",
+    },
+    "token-classification": {
+        "impl": TokenClassificationPipeline,
+        "class": (NeuronModelForTokenClassification,),
+        "default": "dbmdz/bert-large-cased-finetuned-conll03-english",
+        "type": "text",
+    },
+}
+
+
+def load_pipeline(
+    model,
+    targeted_task,
+    load_tokenizer,
+    tokenizer,
+    feature_extractor,
+    load_feature_extractor,
+    SUPPORTED_TASKS,
+    input_shapes=None,
+    export=False,
+    subfolder: str = "",
+    use_auth_token: Optional[Union[bool, str]] = None,
+    revision: str = "main",
+    model_kwargs: Optional[Dict[str, Any]] = None,
+    config: AutoConfig = None,
+    **kwargs,
+):
+    if model_kwargs is None:
+        model_kwargs = {}
+
+    # loads default model
+    if model is None:
+        model_id = SUPPORTED_TASKS[targeted_task]["default"]
+        if input_shapes is None:
+            input_shapes = {"batch_size": 1, "sequence_length": 128}
+            logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+        model = SUPPORTED_TASKS[targeted_task]["class"][0].from_pretrained(model_id, export=True, **input_shapes)
+    # loads model from model id and converts it to neuronx optionally
+    elif isinstance(model, str):
+        model_id = model
+        neuronx_model_class = SUPPORTED_TASKS[targeted_task]["class"][0]
+        model = neuronx_model_class.from_pretrained(model, export=export, **model_kwargs, **input_shapes)
+    # uses neuron model
+    elif isinstance(model, NeuronBaseModel):
+        if tokenizer is None and load_tokenizer:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, (PreTrainedTokenizer, PreTrainedTokenizerFast)):
+                    tokenizer = preprocessor
+                    break
+            if tokenizer is None:
+                raise ValueError(
+                    "Could not automatically find a tokenizer for the NeuronBaseModel, you must pass a tokenizer explictly"
+                )
+        if feature_extractor is None and load_feature_extractor:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, SequenceFeatureExtractor):
```

Comment:
I think this is for speech tasks we do not support yet.

Reply:
yes but code ist mostly copied from original optimum and there is no added value for removing it. so i kept it.